microsoft/testsuites/core/storage.py

```diff
@@ -227,6 +232,40 @@ def verify_resource_disk_io(self, node: RemoteNode) -> None:
             "content read from file should be equal to content written to file",
         ).is_equal_to(original_text)
 
+    @TestCaseMetadata(
+        description="""
+        This test verifies the disk controller type of the VM.
+
+        Steps:
+        1. Get the disk type of the boot partition.
+        2. Compare it with hardware disk controller type of the VM.
+        """,
+        priority=1,
+        requirement=simple_requirement(
+            supported_platform_type=[AZURE],
+        ),
+    )
+    def verify_disk_controller_type(self, node: RemoteNode) -> None:
+        node_disk = node.features[Disk]
+
+        # Get VM's 'disk controller type' with azure api
+        vm_disk_controller_type = node_disk.get_hardware_disk_controller_type()
+
+        # Get 'disk controller type' from within VM.
+        os_disk_controller_type = node_disk.get_os_disk_controller_type()
+
+        # With certain SKUs & gen1 images 'disk_controller_type' will be 'None'
+        if not vm_disk_controller_type:
+            raise SkippedException(
+                "The VM size doesn't have disk controller type information."
+                "Few VM Sizes and gen1 images doesn't support 'disk_controller_type'"
+            )
+
+        assert_that(
+            os_disk_controller_type,
+            "The disk controller types of VM and OS should be the same.",
```

Comment:
vm and os is confusing. Maybe call it like Configured/Runtime?

Reply:
How about hardware and OS ?

Reply:
Yes, it's ok to me.

Reply:
Updated the message with "hardware and OS"